admin/autoupdater/files/usr/sbin/autoupdater

```diff
@@ -257,21 +259,25 @@ local function autoupdate(mirror)
     return true
   end
 
+  autoupdater_util.run_dir('/usr/lib/autoupdater/download.d')
 
-  os.execute('sync; sysctl -w vm.drop_caches=3')
+  autoupdater_util.exec('sync')
+  autoupdater_util.exec('sysctl', '-w', 'vm.drop_caches=3')
   collectgarbage()
 
   local image = os.tmpname()
   if not fetch_firmware(mirror, manifest.filename, image) then
+    autoupdater_util.run_dir('/usr/lib/autoupdater/abort.d')
     return false
   end
 
-  local popen = io.popen(string.format("sha512sum '%s'", image))
+  local popen = io.popen(string.format("exec sha512sum '%s'", image))
```

Comment:
Unrelated to the change at hand, but shouldn't we escape `'` in `image`? Or "simply" reimplement `io.popen()` the way we already have our `autoupdater_utils.popen()`, as passing the command as separate arguments would also solve the problem.

In general, I dislike that now we have some commands that are called as separate arguments and some that are still called via a shell as a single string.